patches/incremental_delivery/0001-Add-prototype-for-opreator-to-iterator.patch

```diff
@@ -0,0 +1,37 @@
+From 8831613cd3cfe9a757909859f9e1e7500fef9ac8 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
+Date: Sun, 23 May 2021 21:08:49 +0200
+Subject: [PATCH] Add prototype for opreator-- to iterator
+
+Otherwise, g++ 10.X refuses to compile this code:
+
+	In file included from /usr/include/c++/10.3.1/bits/stl_algobase.h:66,
+			 from /usr/include/c++/10.3.1/memory:63,
+			 from /home/soeren/src/android-tools/vendor/base/libs/androidfw/include/androidfw/LoadedArsc.h:20,
+			 from /home/soeren/src/android-tools/vendor/base/libs/androidfw/LoadedArsc.cpp:19:
+	/usr/include/c++/10.3.1/bits/stl_iterator_base_funcs.h: In instantiation of 'constexpr void std::__advance(_RandomAccessIterator&, _Distance, std::random_access_iterator_tag) [with _RandomAccessIterator = android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>::const_iterator; _Distance = long int]':
+	/usr/include/c++/10.3.1/bits/stl_iterator_base_funcs.h:206:21:   required from 'constexpr void std::advance(_InputIterator&, _Distance) [with _InputIterator = android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>::const_iterator; _Distance = long int]'
+	/usr/include/c++/10.3.1/bits/stl_algobase.h:1321:16:   required from 'constexpr _ForwardIterator std::__lower_bound(_ForwardIterator, _ForwardIterator, const _Tp&, _Compare) [with _ForwardIterator = android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>::const_iterator; _Tp = short unsigned int; _Compare = __gnu_cxx::__ops::_Iter_comp_val<android::LoadedPackage::GetEntryOffset(android::incfs::verified_map_ptr<android::ResTable_type>, uint16_t)::<lambda(const android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>&, uint16_t)> >]'
+	/usr/include/c++/10.3.1/bits/stl_algo.h:2046:32:   required from 'constexpr _FIter std::lower_bound(_FIter, _FIter, const _Tp&, _Compare) [with _FIter = android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>::const_iterator; _Tp = short unsigned int; _Compare = android::LoadedPackage::GetEntryOffset(android::incfs::verified_map_ptr<android::ResTable_type>, uint16_t)::<lambda(const android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>&, uint16_t)>]'
+	/home/soeren/src/android-tools/vendor/base/libs/androidfw/LoadedArsc.cpp:276:6:   required from here
+	/usr/include/c++/10.3.1/bits/stl_iterator_base_funcs.h:183:2: error: no match for 'operator--' (operand type is 'android::incfs::map_ptr<android::ResTable_sparseTypeEntry, false>::const_iterator')
+	  183 |  --__i;
+	      |  ^~~~~
```

Comment:
No idea what this is about, if anyone has a better idea on how to fix this. Let me know.

Reply:
Hello, **nmeum**. Thank you so much for the update v31.0.2

I added a package fmt-devel (urpmi --auto "pkgconfig(fmt)" - for Mageia Linux only)

mkdir ./111 && cd ./111
git clone --recurse-submodules --depth=1 --branch=platform-tools-31.0.2 https://github.com/nmeum/android-tools.git

cd ./android-tools
mkdir ./build && cd ./build

cmake ..
make

cd ./vendor

./adb version
Android Debug Bridge version 1.0.41
Version 31.0.0p1-android-tools

The android-tool version is still 31.0.0p1. :( I checked - all the patches are applied correctly. In any case, he told me that he was working with v31.0.2 and everything was assembled without errors. In addition, I will leave the address for the already patched archive android-tools-31.0.2.tar.xz: https://cloud.mail.ru/public/gKRe/PJ9JKKZ11

The testing result: the version of android-tools is still 31.0.0p1 and, accordingly, the "adb" problem "Reduce TCP keepalive interval" remains.

I still have little experience with GitHub, maybe I cloned the branch incorrectly? :)

Regarding the error you described above... Unfortunately, I do not program in C++, but I repeat that the build was done using all the patches and without errors. Perhaps I'm not talking about what you need at all, but in the link above, in addition to *.tar.xz, there is also a log file. If I have said too much, please do not judge me too harshly. :)

With respect,
Alex

Reply:
> The android-tool version is still 31.0.0p1.

Yes, didn't update the version string yet. Should be fixed in 111415ed4e71012cca364d6f987726f0407cef92

Reply:
Do you happen to know which upstream commit fixes your TCP timeout issue?